### ISSUE TITLE
syntax-parse: no value found for key 'dots-loop_74

### ISSUE THREAD

**Issue by sorawee** [racket/racket#3470]
**What version of Racket are you using?**
v7.9.0.3 [cs]

**What program did you run?**

```
#lang racket

(require syntax/parse/define)

(begin-for-syntax
  (define-syntax-class clause-pattern
    (pattern :id)))

(define-simple-macro (f [(pat:clause-pattern ...) e ...+] ...)
  1)

(f [(a a) a])
```

**What should have happened?**

No error

**If you got an error message, please include it here.**

```
hash-ref: no value found for key
  key: 'dots-loop_74
  context...:
   /Users/sorawee/git/racket/racket/collects/syntax/wrap-modbeg.rkt:46:4
```

**Please include any other relevant details**

- After switching `clause-pattern` to `id`, the error won't be triggered.
- After switching `...+` to `...`, the error won't be triggered.
- After removing the outermost `...` (that is, make it support exactly one clause), the error won't be triggered.

CC: @rmculpepper

**Comment by sorawee** [comment]
Interestingly, Racket 7.8 BC doesn't seem to have this problem.

**Comment by sorawee** [comment]
Hmm. I can't reproduce the problem in the snapshot versions (both BC and CS). Perhaps this is a fixed bug?

**Comment by sorawee** [comment]
Can't reproduce the problem at HEAD.

**Comment by mflatt** [comment]
Yes, this was fixed by 279412d316ecd78a96b6c8651ed474d43ced1a23.
